app/grandchallenge/components/models.py

```diff
@@ -565,9 +570,32 @@ def file_mimetypes(self):
             )
         elif self.kind == InterfaceKind.InterfaceKindChoices.MP4:
             return ("video/mp4",)
+        elif self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
+            return ("text/x-nh", "application/octet-stream")
         else:
             raise RuntimeError(f"Unknown kind {self.kind}")
 
+    @property
+    def file_extensions(self):
+        exts = [self.file_extension]
+
+        if self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
+            exts.extend([".nwk", ".tree"])
+
+        return exts
+
+    @property
+    def file_extension(self):
+        if self.is_json_kind:
+            return ".json"
+
+        if self.is_file_kind:
+            if self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
+                return ".newick"
+            return f".{self.kind.lower()}"
```

Comment:
Using blocks rather than fall-throughs is preferred, please add this to the style guide:

```suggestion
        elif self.is_file_kind:
            if self.kind == InterfaceKind.InterfaceKindChoices.NEWICK:
                return ".newick"
            else:
                return f".{self.kind.lower()}"
```

Reply:
I have replaced this with hardcorded lookups. I did add it to the style guide.